### PR TITLE
Add OSX build to Travis

### LINKED ISSUE #30: Add OSX to Travis
CRAN requires build success on at least two platforms, so add OSX to the current Travis configuration.

### PR DESCRIPTION
**Pull request by jemrobinson**
Adds OSX build to Travis configuration.

Closes #30.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,16 +1,33 @@
...
+
+# Explicitly enable the lintr-bot for one build (this avoids duplicate comments)
 matrix:
-  include:
-    - r: oldrel
-      env: LINTR_COMMENT_BOT=false
-    - r: release
-    - r: devel
+  exclude:
+    - os: linux
```
Comment (miguelmorin) [inline .travis.yml]: Why do we keep the `exclude` in lines 23-26 now that `LINTR_COMMENT_BOT` is disabled by default?

Comment (jemrobinson) [reply .travis.yml]: Travis treats builds with different environment variables as different builds. We therefore remove the one we don't want in order to replace it with the one we do. I've tried using conditions to change this variable, but I don't think it works - will revert.

File: .travis.yml [hunk 2]
```diff
@@ -1,16 +1,33 @@
...
+# Construct the matrix of OS and R versions
+os:
+  - linux
+  - osx
+r:
+  - oldrel
+  - release
+  - devel
+
+# Explicitly enable the lintr-bot for one build (this avoids duplicate comments)
```
Comment (miguelmorin) [inline .travis.yml]:
```suggestion
# Explicitly enable the lintr-bot for one build (this avoids duplicate comments).
# Travis treats builds with different environment variables as different builds.
# We therefore remove the one we don't want in order to replace it with the one
# we do.
```

**Comment by miguelmorin** [comment]
I expanded on the comment and updated the branch with my other pull requests. Ready for @jemrobinson to rebase and merge.

**Pull request merged**